YAKL_reductions.h

```diff
@@ -232,6 +232,105 @@
       }
     };
 
+  #elif defined(__USE_SYCL__)
+
+    template <class T> class ParallelMin<T,memDevice> {
+      int    nItems; // Number of items in the array that will be reduced
+      T      *rsltP; // Device pointer for reduction result
+      public:
+      ParallelMin() { }
+      ParallelMin(int const nItems) { setup(nItems); }
+      ~ParallelMin() { finalize(); }
+      void setup(int const nItems) {
+        finalize();
+        rsltP = (T *) yaklAllocDevice(sizeof(T),""); // Allocate device pointer for result
+        this->nItems = nItems;
+      }
+
+      void finalize() {
+        yaklFreeDevice(rsltP,"");
+      }
+      T operator() (T *data) {
+        T rslt;
+        sycl_default_stream.memcpy(&rslt,rsltP,sizeof(T)); // Copy result to host
+        return rslt;
+      }
+      void deviceReduce(T *data, T *devP) {
+        sycl_default_stream.submit([&](cl::sycl::handler& cgh) {
+          auto minReduction = cl::sycl::ONEAPI::reduction(devP, cl::sycl::ONEAPI::minimum<>());
+            cgh.parallel_for(cl::sycl::nd_range<1>{nItems, 32},
+                             minReduction,
+                             [=](cl::sycl::nd_item<1> idx, auto& min) {
+			       min.combine(data[idx.get_global_id()]);
+                             });
```

Comment:
nItems is the # items being reduced. 32 is the size of the "block" (in the future, we'll want to let this respond to the type of Intel hardware).